File: graphs/binary_tree_level_order_traversal.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def levelOrder(self, root: Optional[TreeNode]) -> List[List[int]]:
        queue = []

        queue.append(root)
        result = []

        while queue:
            level = []
            n = len(queue)

            for nums in range (n)
                curr = queue.popleft

                if curr is empty:
                    return next

                level.append(curr.val)
                
                if curr.left is not None:
                    queue.append(curr.left)
                if curr.right is not None:
                    queue.append(curr.right)

            if level:
                result.append(level)

        return result

File: graphs/binary_tree_level_order_traversal.rb
# Definition for a binary tree node.
# class TreeNode
#     attr_accessor :val, :left, :right
#     def initialize(val = 0, left = nil, right = nil)
#         @val = val
#         @left = left
#         @right = right
#     end
# end
# @param {TreeNode} root
# @return {Integer[][]}
def level_order(root)
    queue = []

    queue << root
    result = []

    while !queue.empty?
        level = []

        queue.size.times do
            curr = queue.shift

            next if curr.nil?

            level << curr.val
            
            queue << curr.left if !curr.left.nil?
            queue << curr.right if !curr.right.nil?
        end
        result << level if !level.empty?
    end
    result
end